src/Compiler/Utilities/TaggedCollections.fs

```diff
@@ -987,95 +1193,123 @@ module MapTree =
+        refresh m (MapTree.remove comparer k tree)
+
     member m.TryFind(k) = MapTree.tryFind comparer k tree
     member m.ToList() = MapTree.toList tree
     member m.ToArray() = MapTree.toArray tree
 
-    static member FromList(comparer : 'ComparerTag,l) : Map<'Key,'T,'ComparerTag> = 
-        Map<_,_,_>(comparer=comparer, tree=MapTree.ofList comparer l)
+    static member FromList(comparer: 'ComparerTag, l) : Map<'Key, 'T, 'ComparerTag> =
+        Map<_, _, _>(comparer = comparer, tree = MapTree.ofList comparer l)
 
-    static member Create(comparer : 'ComparerTag, ie : seq<_>) : Map<'Key,'T,'ComparerTag> = 
-        Map<_,_,_>(comparer=comparer, tree=MapTree.ofSeq comparer ie)
+    static member Create(comparer: 'ComparerTag, ie: seq<_>) : Map<'Key, 'T, 'ComparerTag> =
+        Map<_, _, _>(comparer = comparer, tree = MapTree.ofSeq comparer ie)
 
     interface IEnumerable<KeyValuePair<'Key, 'T>> with
         member s.GetEnumerator() = MapTree.toSeq tree
 
     interface System.Collections.IEnumerable with
-        override s.GetEnumerator() = (MapTree.toSeq tree :> System.Collections.IEnumerator)
+        override s.GetEnumerator() =
+            (MapTree.toSeq tree :> System.Collections.IEnumerator)
 
-    override this.Equals(that) = 
+    override this.Equals(that) =
         match that with
         // Cast to the exact same type as this, otherwise not equal.
-        | :? Map<'Key,'T,'ComparerTag> as that -> ((this :> System.IComparable).CompareTo(that) = 0)
+        | :? Map<'Key, 'T, 'ComparerTag> as that -> ((this :> System.IComparable).CompareTo(that) = 0)
         | _ -> false
 
-    interface System.IComparable with 
-         member m1.CompareTo(m2: obj) = 
-             Seq.compareWith 
-               (fun (kvp1 : KeyValuePair<_,_>) (kvp2 : KeyValuePair<_,_>)-> 
-                   let c = m1.Comparer.Compare(kvp1.Key,kvp2.Key) in 
-                   if c <> 0 then c else Unchecked.compare kvp1.Value kvp2.Value)
-               // Cast m2 to the exact same type as m1, see 4884.
-               // It is not OK to cast m2 to seq<KeyValuePair<'Key,'T>>, since different compares could permute the KVPs.
-               m1 (m2 :?> Map<'Key,'T,'ComparerTag>)
-
-    member this.ComputeHashCode() = 
-        let combineHash x y = (x <<< 1) + y + 631 
+    interface System.IComparable with
+        member m1.CompareTo(m2: obj) =
+            Seq.compareWith
+                (fun (kvp1: KeyValuePair<_, _>) (kvp2: KeyValuePair<_, _>) ->
+                    let c = m1.Comparer.Compare(kvp1.Key, kvp2.Key) in
+
+                    if c <> 0 then
+                        c
+                    else
+                        Unchecked.compare kvp1.Value kvp2.Value)
+                // Cast m2 to the exact same type as m1, see 4884.
+                // It is not OK to cast m2 to seq<KeyValuePair<'Key,'T>>, since different compares could permute the KVPs.
+                m1
+                (m2 :?> Map<'Key, 'T, 'ComparerTag>)
```

Comment:
That looks a bit odd, `m1` on its own line.

Reply:
The function application `Seq.compareWith` takes three arguments and cannot be formatted on one line. See https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-application-expressions
Nothing odd here.

Reply:
Oh, you're righ. Big compare function confused me a bit. We should probably bind it to something and use as named function, should be easier to read.